src/pages/ShowFeelingsPage/ShowFeelingTemplate.tsx

```diff
@@ -1,59 +1,224 @@
+                  <i>
+                    { (showOptions
+                      && (selectedFeeling === feelingsListObject[Number(feelingId)].id))
+                      ? <ChevronDown />
+                      : <ChevronRight />}
+                  </i>
+                  <br />
+                  {feelingsListObject[Number(feelingId)]?.note && (<span className="btn-my-feelings__note">...</span>)}
+                  {showOptions
+                  && (selectedFeeling === feelingsListObject[Number(feelingId)].id) && (
+                    <div className="show-feelings__options">
+                      <TrashFill
+                        onClick={() => {
+                          const numFeelingId = feelingsListObject[Number(feelingId)].id;
+                          if (numFeelingId !== undefined) { removeFeeling(numFeelingId); } else { console.log('Attempting to remove feeling not in the database'); }
+                          const newFeelingsList = currentFeelingsList;
+                          const feelingDate = feelingsListObject[Number(feelingId)].date;
+                          newFeelingsList[feelingDate] = currentFeelingsList[feelingDate].filter(
+                            (feelingOnDate: IFeelingItem) => feelingOnDate.id !== numFeelingId,
+                          );
+                          setFeelingsListObject.setFeelingsList({ ...newFeelingsList });
+                        }}
+                        size={20}
+                      />
+                      <Journal
+                        onClick={() => {
+                          if (feelingsListObject[Number(feelingId)]?.note) {
+                            setSelectedFeelingNote(feelingsListObject[Number(feelingId)]?.note!);
+                          }
+                          // eslint-disable-next-line no-unused-expressions
+                          feelingsListObject[Number(feelingId)]?.note
+                            ? handleNotesShow()
+                            : handleInputShow();
+                        }}
+                        size={20}
+                      />
+                    </div>
+                  )}
+                </Button>
+              )),
+            )}
+            <Modal show={showInputModal} onHide={handleInputClose} centered>
+              <Modal.Header closeButton>
+                <Modal.Title><i>Want to tell more about it?</i></Modal.Title>
+              </Modal.Header>
+              <Modal.Body>
+                <input
+                  type="text"
+                  placeholder="Add a reason"
+                  className="show-feelings__note-input"
+                  value={noteValue}
+                  onChange={(e) => { setNoteValue(e.target.value); }}
+                />
+              </Modal.Body>
+              <Modal.Footer>
+                <Button
+                  variant="primary"
+                  onClick={() => {
+                    addFeelingNote(selectedFeeling!, noteValue);
```

Comment:
Well, When you are adding note, you are directly updating the Dexi DB. My suggestion is after updating the DB return the updated feelings from the DB. So update the API call after adding the note in DB return the updated data from there and change the state.

Reply:
Yes that's what I tried before but did not get the re-rendering successful. Let me go back to that commit and post it here for your reference